tests/e2e/default/more-options-menu/offline-user.wdio-spec.js

```diff
@@ -0,0 +1,196 @@
+  isOffline: true,
+  roles:['chw'],
+  place: health_center._id,
+  contact: contact._id,
+});
+
+
+const patient = personFactory.build({
+  _id: uuid(),
+  parent: { _id: clinic._id, parent: { _id: health_center._id, parent: { _id: district_hospital._id }}}
+});
+
+const reports = [
+  reportFactory.build(
+    {
+      form: 'P',
+      reported_date: moment([ today.year(), today.month() - 4, 1, 23, 30 ]).valueOf(),
+      patient_id: patient._id,
+    },
+    {
+      patient, submitter: offlineUser.contact, fields: { lmp_date: 'Feb 3, 2022', patient_id: patient._id},
+    },
+  ),
+];
+
+const sendMessage = async (message = 'Testing', phone = contact.phone) => {
+  await utils.request({
+    method: 'POST',
+    path: '/api/v2/records',
+    headers: {
+      'Content-type': 'application/x-www-form-urlencoded'
+    },
+    body:`message=${message}&from=${phone}`,
+  });  
+};
+
+const updatePermissions = async (role, addPermissions, removePermissions = []) => {
+  const settings = await utils.getSettings();
+  addPermissions.map(permission => settings.permissions[permission].push(role));
+  removePermissions.forEach(permission => {
+    settings.permissions[permission] = [''];
+  });
+  await utils.updateSettings({ roles: settings.roles, permissions: settings.permissions }, true);
+};
+
+describe('More Options Menu - Offline User', async () => {
+  before(async () => {
+    await utils.saveDocs([ ...places.values(), contact, patient, ...reports ]);
+    await utils.createUsers([offlineUser]);
+    await loginPage.login(offlineUser);
+  });
+
+  describe('all permissions enabled', async () => {
+    it('- Message tab', async () => {
+      await commonPage.goToMessages();
+      await sendMessage();
+      expect(await (await commonPage.moreOptionsMenu()).isExisting()).to.be.false;    
+    });
+
+    it('- Contact tab: no contact selected', async () => {
+      await commonPage.goToPeople();
+      //parent contact
+      await commonPage.openMoreOptionsMenu();
+      expect(await commonPage.isOptionVisible('export', 'contacts')).to.be.false;
+      expect(await commonPage.isOptionVisible('edit', 'contacts')).to.be.false;
+      expect(await commonPage.isOptionEnabled('delete', 'contacts')).to.be.false;
+    });
+
+    it(' - Contact Tab : contact selected', async () => {
+      await commonPage.goToPeople();
+      //contact selected
+      await contactPage.selectLHSRowByText(contact.name);
+      await commonPage.openMoreOptionsMenu();
+      expect(await commonPage.isOptionVisible('export', 'contacts')).to.be.false;
+      expect(await commonPage.isOptionEnabled('edit', 'contacts')).to.be.true;
+      expect(await commonPage.isOptionEnabled('delete', 'contacts')).to.be.true;
+    });
+
+    it('- options enabled when report selected', async () => {
+      await commonPage.goToReports();
+      expect(await (await commonPage.moreOptionsMenu()).isExisting()).to.be.false;    
+      (await reportPage.firstReport()).click();
```

Comment:
Is it selecting a XML or a No XML report?
Because since you added both and DB isn't cleared, how do we know which one it is testing?

We should test both and check the display of the Edit